grouper/permissions.py

```diff
@@ -50,6 +51,87 @@ def __init__(self, name):
         self.name = name
 
 
+def create_permission(session, name, description=''):
+    # type: (Session, str, Optional[str]) -> Permission
+    """Create and add a new permission to database
+
+    Arg(s):
+        session(models.base.session.Session): database session
+        name(str): the name of the permission
+        description(str): the description of the permission
+
+    Returns:
+        The created permission that has been added to the session
+    """
+    permission = Permission(name=name, description=description or '')
+    permission.add(session)
+    return permission
+
+
+def get_all_enabled_permissions(session):
+    # type: (Session) -> List[Permission]
+    """Get all enabled permissions
+
+    Arg(s):
+        session(models.base.session.Session): database session
+
+    Returns:
+        List of all enabled permissions, sorted by ascending permission name
+    """
+    return session.query(
+        Permission,
+    ).filter(
+        Permission.enabled == True,
+    ).order_by(asc(Permission.name)).all()
+
+
+def get_all_permissions(session):
+    # type: (Session) -> List[Permission]
+    """Get all permissions, enabled or disabled
+
+    Arg(s):
+        session(models.base.session.Session): database session
+
+    Returns:
+        List of all permissions---enabled or not---sorted by ascending permission name
+    """
+    return session.query(Permission).order_by(asc(Permission.name)).all()
+
+
+def get_permission(session, name):
```

Comment:
i think this is fine because i dont think people care about the permission directly, but rather people are interested in things like, `user_has_permission()`, `get_groups_by_permission()`, etc. and with the most recent commit (just moments ago), i've added a test that makes sure all those api's do the right thing, i.e., skip disabled permissions.